src/gkmodel/node_model.py

```diff
@@ -232,17 +236,32 @@ def predict(self,
             for study_id in data.study_id
         ])
 
+        adjust_values = np.zeros(self.data.num_points)
+
         if slope_quantile is not None:
-            for name, quantile in slope_quantile.items():
-                if name in self.cov_names:
-                    i = self.cov_names.index(name)
-                    v = np.quantile(soln[:, i], quantile)
-                    if quantile >= 0.5:
-                        soln[:, i] = np.maximum(soln[:, i], v)
-                    else:
-                        soln[:, i] = np.minimum(soln[:, i], v)
-
-        return np.sum(mat*soln, axis=1)
+            covs_index = [self.cov_names.index(name) for name, _ in slope_quantile.items() if name in self.cov_names]
```

Comment:
for `covs_index` and `quantiles`, it might be better to store the `cov_names` in `slope_quantiles` that is useful as a list, and construct the `covs_index` and `quantiles` from it, since I feel a little uncertainty that `slope_quantile.items()` always have the same order.

Reply:
I think Python dictionary is order-preserving, but I certainly can change that just to be safe.

Reply:
Thanks very much!